pkg/metrics/listener.go

```diff
@@ -44,9 +54,36 @@ func NewListener(addr string) (net.Listener, error) {
 	log.Info("metrics server is starting to listen", "addr", addr)
 	ln, err := net.Listen("tcp", addr)
 	if err != nil {
-		er := fmt.Errorf("error listening on %s: %w", addr, err)
-		log.Error(er, "metrics server failed to listen. You may want to disable the metrics server or use another port if it is due to conflicts")
-		return nil, er
+		err = fmt.Errorf("error listening on %s: %w", addr, err)
+		log.Error(err, "metrics server failed to listen. You may want to disable the metrics server or use another port if it is due to conflicts")
+		return nil, err
 	}
+
 	return ln, nil
 }
+
+// NewSecureListener creates a new TCP listener over TLS bound to the given address.
+func NewSecureListener(addr, certDir string) (net.Listener, error) {
+	ln, err := NewListener(addr)
```

Comment:
This error is unchecked
```suggestion
	ln, err := NewListener(addr)
	if err != nil {
	  return nil, err
	}
```